testing/e2e/scaffold/scaffold.go

```diff
@@ -511,34 +511,48 @@ func (s *Scaffold) FleetHasArtifacts(ctx context.Context) []ArtifactHit {
 	}
 }
 
-func (s *Scaffold) StartToxiproxy(ctx context.Context) *toxiproxy.Client {
-	req := testcontainers.ContainerRequest{
-		Image:        "ghcr.io/shopify/toxiproxy:2.5.0",
-		ExposedPorts: []string{"8474/tcp"},
-		WaitingFor:   wait.ForHTTP("/version").WithPort("8474/tcp"),
-		NetworkMode:  "host",
-	}
-	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
-		ContainerRequest: req,
-		Started:          true,
-	})
+type logger struct {
+	*testing.T
+}
+
+func (l *logger) Printf(format string, v ...interface{}) {
+	l.Helper()
+	l.Logf(format, v...)
+}
+
+func (s *Scaffold) StartToxiproxy(ctx context.Context) *toxitc.Container {
+	container, err := toxitc.Run(ctx, "ghcr.io/shopify/toxiproxy:2.12.0",
+		testcontainers.CustomizeRequest(testcontainers.GenericContainerRequest{
+			ContainerRequest: testcontainers.ContainerRequest{
+				Hostname: "toxi",
+				Name:     "toxi",
+				// Network is set to the integration test network instead of using host mode so it can easily communicate with other containers.
+				// NOTE: the container will not become healthy when using `testcontainers-go 0.36.x+ if set to NetworkMode: "host"
+				Networks: []string{"integration_default"},
+			}}),
+		testcontainers.WithLogger(&logger{s.T()}),
+		toxitc.WithProxy("es", "elasticsearch:9200"),
+	)
 	s.Require().NoError(err)
 
 	s.T().Cleanup(func() {
+		if s.T().Failed() {
+			rc, err := container.Logs(context.Background())
```

Comment:
Should we use a context with a timeout here, so we're not potentially waiting forever?